tests/core/test_trainer.py

```diff
@@ -122,3 +124,60 @@ def test_from_argparse_args():
     trainer = Trainer.from_argparse_args(args)
     assert trainer.max_epochs == 200
     assert isinstance(trainer, Trainer)
+
+
+@pytest.mark.parametrize("stage", [RunningStage.TRAINING, RunningStage.VALIDATING, RunningStage.TESTING])
+def test_trainer_request_dataloaders_legacy(stage):
+    """Test to ensure that ``request_dataloaders`` can take the legacy PL ordering of arguments.
+
+    legacy: (model, stage)
+    """
+
+    class TestTrainer(Trainer):
+        recorded_on_dataloader_calls = {}
+
+        def on_train_dataloader(self) -> None:
+            self.recorded_on_dataloader_calls[RunningStage.TRAINING] = True
+
+        def on_val_dataloader(self) -> None:
+            self.recorded_on_dataloader_calls[RunningStage.VALIDATING] = True
+
+        def on_test_dataloader(self) -> None:
+            self.recorded_on_dataloader_calls[RunningStage.TESTING] = True
```

Comment:
I think stage was a string in the old usage
```suggestion
        def on_train_dataloader(self) -> None:
            self.recorded_on_dataloader_calls["train"] = True

        def on_val_dataloader(self) -> None:
            self.recorded_on_dataloader_calls["val"] = True

        def on_test_dataloader(self) -> None:
            self.recorded_on_dataloader_calls["test"] = True
```